dockerunit-consul/src/main/java/com/github/qzagarese/dockerunit/discovery/consul/annotation/impl/UseConsulDnsExtensionInterpreter.java

```diff
@@ -1,18 +1,48 @@
 package com.github.qzagarese.dockerunit.discovery.consul.annotation.impl;
 
-import static com.github.qzagarese.dockerunit.discovery.consul.ConsulDiscoveryConfig.DOCKER_BRIDGE_IP_DEFAULT;
-import static com.github.qzagarese.dockerunit.discovery.consul.ConsulDiscoveryConfig.DOCKER_BRIDGE_IP_PROPERTY;
-
 import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.command.ListContainersCmd;
+import com.github.dockerjava.api.model.Container;
+import com.github.dockerjava.api.model.ContainerNetwork;
 import com.github.qzagarese.dockerunit.annotation.ExtensionInterpreter;
+import com.github.qzagarese.dockerunit.discovery.consul.ConsulDiscoveryConfig;
 import com.github.qzagarese.dockerunit.discovery.consul.annotation.UseConsulDns;
 import com.github.qzagarese.dockerunit.internal.ServiceDescriptor;
+import com.github.qzagarese.dockerunit.internal.docker.DefaultDockerClientProvider;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static com.github.qzagarese.dockerunit.discovery.consul.ConsulDiscoveryConfig.DOCKER_BRIDGE_IP_DEFAULT;
+import static com.github.qzagarese.dockerunit.discovery.consul.ConsulDiscoveryConfig.DOCKER_BRIDGE_IP_PROPERTY;
 
 public class UseConsulDnsExtensionInterpreter implements ExtensionInterpreter<UseConsulDns> {
 
+
+    private final com.github.dockerjava.api.DockerClient dockerClient;
+
+    public UseConsulDnsExtensionInterpreter() {
+        dockerClient = new DefaultDockerClientProvider().getClient();
+    }
+
     @Override
     public CreateContainerCmd build(ServiceDescriptor sd, CreateContainerCmd cmd, UseConsulDns t) {
-        return cmd.withDns(System.getProperty(DOCKER_BRIDGE_IP_PROPERTY, DOCKER_BRIDGE_IP_DEFAULT));
+        ListContainersCmd listCmd = dockerClient.listContainersCmd();
+        Map<String, List<String>> filters = listCmd.getFilters();
+        filters.put("name", Arrays.asList(ConsulDiscoveryConfig.CONSUL_CONTAINER_NAME));
```

Comment:
the above could be one statement (`filters` isn't used after)
```
listCmd.getFilters().put("name", Arrays.asList(ConsulDiscoveryConfig.CONSUL_CONTAINER_NAME));
```

Reply:
I want to remove this anyway because as it stands it's a hack and relies too much on the internal implementation of `docker-java`.

I rather filter on the result and then reintroduce it once 3.1.1 comes out as stable.